website/docs/language/modules/develop/refactoring.html.md

```diff
@@ -0,0 +1,461 @@
+module call used `count` or `for_each` then it would apply to all of the
+instances, without the need to specify each one separately.
+
+## Enabling `count` or `for_each` For a Module Call
+
+Imagine that the initial version of your module contains a single-instance
+module:
+
+```hcl
+module "a" {
+  source = "../modules/example"
+
+  # (module arguments)
+}
+```
+
+Applying this configuration would cause Terraform to create objects whose
+addresses begin with `module.a`.
+
+Later requirements might lead to you needing to use
+[`count`](../../meta-arguments/count.html) with this resource, in order
+to systematically declare multiple instances. To preserve an object that
+was previously associated with `aws_instance.a` alone, you can used a `moved`
+block to specify which instance key that object will take in the new
+configuration:
+
+```hcl
+module "a" {
+  source = "../modules/example"
+  count  = 3
+
+  # (module arguments)
+}
+
+moved {
+  from = module.a
+  to   = module.a[2]
+}
+```
+
+The above will cause Terraform to treat all objects in `module.a` as if they
+were originally created in `module.a[2]`, and thus it will plan to create
+new objects only for `module.a[0]` and `module.a[1]`.
+
+When at least one of the two addresses includes an instance key, like
+`[2]` in the above example, Terraform will understand both addresses as
+referring to specific _instances_ of a module call rather than the module
+call as a whole. That means you can use `moved` to switch between keys and to
+add and remove keys as you switch between `count`, `for_each`, or neither.
+
+For more examples of recording moves associated with instances, refer to
+the similar section
+[Enabling `count` and `for_each` For a Resource](#enabling-count-or-for_each-for-a-resource).
+
+# Splitting One Module into Multiple
+
+As a module grows to support new requirements, it might eventually grow big
+enough to warrant splitting into two separate modules.
+
+Consider the following initial version of a module:
+
+```hcl
+resource "aws_instance" "a" {
+  # (other resource-type-specific configuration)
+}
+
+resource "aws_instance" "b" {
+  # (other resource-type-specific configuration)
+}
+
+resource "aws_instance" "c" {
+  # (other resource-type-specific configuration)
+}
+```
+
+You can split this into two modules as follows:
+
+* `aws_instance.a` will now belong to module "x".
+* `aws_instance.b` will also belong to module "x".
+* `aws_instance.c`, though, will belong to module "y".
+
+To achieve this refactoring without replacing existing objects bound to the
+old resource addresses, you must:
+
+1. Write module "x", copying over the two resources it should contain.
+2. Write module "y", copying over the one resource it should contain.
+3. Edit the original module to no longer include any of these resources, and
+   instead to contain only shim configuration to migrate existing users.
+
+The new modules "x" and "y" will therefore contain only `resource` blocks,
+as normal:
```

Comment:
One last thought - you use "as normal" a couple of times I think and I'm not 100% sure what it means. I wonder if you can just omit it as I've suggested here. 

```suggestion
The new modules "x" and "y" should contain only `resource` blocks:
```

Reply:
With some time away from this _I'm_ honestly not sure what I meant by "as normal" here either, and I agree it doesn't really seem to be adding anything here.

(I _guess_ I was probably trying to say that there are no `moved` blocks in the child modules, because in this case they appear only in the root module, but another paragraph further down gets that message across more directly anyway. :man_shrugging: )